File: 2st/Yeonwoo/빈 배열에 추가, 삭제하기.py
def solution(arr, flag):
    answer = []
    temp = []
    for i, tf in enumerate(flag):
        if tf==True:
            for j in range(1,arr[i]*2+1):
                answer.append(arr[i])
        else:
            answer = answer[:-arr[i]]
    
    return answer

File: 4st/Hani/programmer/빈 배열에 추가, 삭제하기.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        List<Integer> list = new ArrayList<>();
        
        for (int i = 0 ; i < arr.length; i++) {
            if (flag[i] == true) {
                for (int j = 0; j < arr[i]*2; j++) {
                     list.add(arr[i]);
                }
            }
            else if (flag[i] == false) {
                 for (int j = 0; j < arr[i]; j++) {
                     list.remove(list.size() - 1);
                }
            }
        }
        
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}